[John Russell Himawan] iP

johnrhimawan opened the pull request:
# DukePro

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER FAST* to use

All you need to do is,
1. download it from ([here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you're a Java programmer, you can use it to practice Java too. Here's the ``` main``` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

johnrhimawan committed:
Add greet, echo, exit functionality

johnrhimawan committed:
Add add,list feature

johnrhimawan committed:
Add indentation in displayList

johnrhimawan committed:
Add mark and unmark feature

johnrhimawan committed:
Add todo, event, deadline tasks

johnrhimawan committed:
Add automated text ui testing

johnrhimawan committed:
Add exception handling

johnrhimawan committed:
Add delete feature

johnrhimawan committed:
Add makeIndent method

johnrhimawan committed:
Remove extra space

johnrhimawan committed:
Edit expected result

johnrhimawan committed:
Add test case

johnrhimawan committed:
Edit chatbot personality

johnrhimawan committed:
Add write file feature

johnrhimawan committed:
Add date format and suitable exception

johnrhimawan committed:
Merge branch 'branch-Level-8' into master

johnrhimawan committed:
Add more OOP

johnrhimawan committed:
Add packages

johnrhimawan committed:
Add gradle support

johnrhimawan committed:
Add gradle support

Add gradle support

johnrhimawan committed:
Delete deprecated Duke.java file

johnrhimawan committed:
Organize into packages

johnrhimawan committed:
Fix Todo name

johnrhimawan committed:
Add parser and task test cases

lesterong commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,36 @@
+import java.time.format.DateTimeParseException;
+
+public class EventCommand extends Command {
+    String description;
+    String at;
```
Should these instance variables be private?

LJXSean replied:
I think so too @lesterong

lesterong commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeMissingSpecifierException, DukeInvalidDateException {
```
Perhaps this line could be broken up into two shorter lines to improve readability.

lesterong commented on src/main/java/duke/exception/DukeInvalidDateException.java:
```diff
@@ -0,0 +1,8 @@
+
+public class DukeInvalidDateException extends DukeException {
+    public DukeInvalidDateException() {
+        super("It seems like you are using an invalid date format. Please input your date in the format " +
+                "yyyy-mm-dd");
```
Should the `+` operator be placed together with the second line instead?

lesterong commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,95 @@
+package duke.parser;
+
+import duke.command.*;
```
Perhaps listing each of the imports separately might make the code easier to comprehend and maintain.

lesterong commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,93 @@
+import duke.task.Todo;
+
+public class Storage {
+    private static final String DIR = System.getProperty("user.dir");
```
Since the scope of this variable is relatively wide, should this variable have a longer and more descriptive name like `DIRECTORY`?

lesterong commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    public void displayGreeting() {
+        printMessage(makeIndent("Hi, how are you doing?! I'm JRH2000\n") +
+                makeIndent("How can I help you?\n"));
```
Perhaps the `+` operator can be placed on the second line instead to improve readability.

lesterong reviewed (commented):
Hey @johnrhimawan, overall, I think you've done a good job so far, with a minor nitpicks here and there. Good job!

johnrhimawan committed:
Update build.gradle

johnrhimawan committed:
Add javadocs

johnrhimawan committed:
Merge branch 'branch-A-JavaDoc' into master

johnrhimawan committed:
Add coding standard

johnrhimawan committed:
Merge tag 'A-CodingStandard' into master

johnrhimawan committed:
Add find

johnrhimawan committed:
Merge branch 'Level-9' into master

johnrhimawan committed:
Add checkstyle

johnrhimawan committed:
Add GUI feature

johnrhimawan committed:
Add GUI feature

johnrhimawan committed:
Merge branch 'branch-Level-10' into master

LJXSean commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,48 @@
+import duke.task.TaskList;
+import duke.ui.Ui;
+import duke.exception.DukeOutOfBoundsException;
+import duke.task.Task;
```
Perhaps the ordering of import statements could be more consistent with other classes to make it easier to browse the list?

johnrhimawan committed:
Add exit application feature

LJXSean commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
```
Perhaps you could elaborate on the name to explain what is being flipped?

LJXSean commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,49 @@
+        }
+        Task deletedTask = tasks.deleteTask(index);
+        storage.save(tasks);
+        return ui.displayDelete(deletedTask);
```
Would displayDeletedTask be a better name to avoid ambiguity even though it is longer?

johnrhimawan committed:
Add greeting GUI

LJXSean commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,118 @@
+            if (splitCommand.length < 2) {
+                throw new DukeEmptyDescriptionException("deadline");
+            }
+            String desc = splitCommand[1];
```
Since the full spelling is being used everywhere else, perhaps the full spelling of description would be better instead of using an abbreviation?

LJXSean reviewed (commented):
LGTM other than some minor nits in naming code quality

johnrhimawan committed:
Update image

johnrhimawan committed:
Add assertion in TaskList

johnrhimawan committed:
Merge branch 'branch-A-Assertions' into master

johnrhimawan committed:
Fix code quality issues

johnrhimawan committed:
Merge branch 'branch-A-CodeQuality' into master

johnrhimawan committed:
Add remind task functionality

Reminding tasks is a new feature implemented into this program. It
is used to tell the user which tasks need to be completed within
7 days.

This feature can be used when the user types in "remind" in the
CLI.

The goal of this command is to assist users to easily view which
tasks need to be completed within 7 days.

johnrhimawan committed:
Improve code quality

johnrhimawan committed:
Remove unused import

johnrhimawan committed:
Improve code quality

johnrhimawan committed:
Add exception description

johnrhimawan committed:
Merge pull request from johnrhimawan/branch-A-MoreErrorHandling

Add exception description

johnrhimawan committed:
Merge branch 'master' into branch-A-Assertions

johnrhimawan committed:
Add more tests

johnrhimawan committed:
Add more testing

johnrhimawan committed:
Merge pull request #4 from johnrhimawan/branch-A-MoreTesting

Add more testing

johnrhimawan committed:
Update README

johnrhimawan committed:
Add Ui.png

johnrhimawan committed:
Merge branch 'branch-A-Assertions' into master

johnrhimawan committed:
Improve code quality

johnrhimawan committed:
Add text bubble

johnrhimawan committed:
Add feature to wait a few seconds before exiting program

johnrhimawan committed:
Update Ui.png

johnrhimawan committed:
Fix bug in bye command

johnrhimawan committed:
Fix formatting
